components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/utils/ServiceCatalogUtils.java

```diff
@@ -85,17 +87,31 @@
     private static String updateServiceUrl(String currentUrl) throws ResolverException {
         /*
             Supported formats
-            https://{host}:{port}/api1
-            https://{url}/api1
+            https://{MI_HOST}:{MI_PORT}/api1
+            https://{MI_URL}/api1
         */
-        SystemResolver resolver = new SystemResolver();
-        if (currentUrl.contains(HOST) && currentUrl.contains(PORT)) {
-            resolver.setVariable(MI_HOST);
-            String host = resolver.resolve();
-            resolver.setVariable(MI_PORT);
-            String port = resolver.resolve();
-            currentUrl = currentUrl.replace(HOST, host).replace(PORT, port);
+        if (currentUrl.contains(HOST) || currentUrl.contains(PORT)) {
+            Map<String, Object> configParser = ConfigParser.getParsedConfigs();
+            String envHost = System.getenv(MI_HOST);
+            // update the {MI_HOST} with server.hostname value
+            if (currentUrl.contains(HOST) && (envHost == null || envHost.isEmpty())) {
+                envHost = (String) configParser.get(SERVER_HOSTNAME);
+            }
+            String envPort = System.getenv(MI_PORT);
+            // update the {MI_PORT} with server listener ports + offset values
+            if (currentUrl.contains(PORT) && (envPort == null || envPort.isEmpty())) {
+                int portOffset = Integer.parseInt(System.getProperty(SERVER_PORT_OFFSET));
+                if (currentUrl.startsWith("https")) {
+                    int httpsPort = Integer.parseInt((String) configParser.get(HTTPS_LISTENER_PORT));
+                    envPort = String.valueOf(httpsPort + portOffset);
+                } else {
+                    int httpPort = Integer.parseInt((String) configParser.get(HTTP_LISTENER_PORT));
+                    envPort = String.valueOf(httpPort + portOffset);
+                }
+            }
+            currentUrl = currentUrl.replace(HOST, envHost).replace(PORT, envPort);
         } else if (currentUrl.contains(URL)) {
+            SystemResolver resolver = new SystemResolver();
```

Comment:
Why have we used the SystemResolver for this and pure Syste.getEnv for the previous case? Shall we be consistent?

Reply:
In previous cases, it checks the `MI_HOST` and `MI_PORT` null check to fetch the system default values (to avoid breaking default behaviour). But for `MI_URL` we don't need to resolve it by default since it is a manually configured placeholder.

Reply:
Assume a day where we have changed the SystemResolver logic to lookup system properties as well. The 2 cases will behave differently. So let's make it consistent.